packages/slate-editor/src/extensions/button-block/components/ButtonBlockElement.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useCallback } from 'react';
+import { useSlateStatic, type RenderElementProps } from 'slate-react';
+
+import { EditorBlock } from '#components';
+
+import type { ButtonBlockNode } from '../ButtonBlockNode';
+import { removeButtonBlock, updateButtonBlock } from '../transforms';
+
+import { Button } from './Button/Button';
+import { ButtonMenu, type FormState } from './ButtonBlockMenu';
+
+interface Props extends RenderElementProps {
+    element: ButtonBlockNode;
+    withNewTabOption: boolean;
+}
+
+export function ButtonBlockElement({ attributes, children, element, withNewTabOption }: Props) {
+    const editor = useSlateStatic();
+
+    const { layout } = element;
+
+    const align = layout === 'wide' ? 'center' : layout;
+
+    const handleUpdate = useCallback(
+        function (patch: Partial<FormState>) {
+            updateButtonBlock(editor, patch);
+        },
+        [editor, element],
+    );
+
+    const handleRemove = useCallback(
+        function () {
+            removeButtonBlock(editor);
+        },
```

Comment:
Let's pass the element to `removeButtonBlock()` to make sure it removes the right one.

Check `removeCoverage` for example: https://github.com/prezly/slate/blob/75814cf69617c3fe4d6a26535ab6e275ebd4de8a/packages/slate-editor/src/extensions/coverage/components/CoverageElement.tsx#L49-L53